### ISSUE TITLE
Assets: bring credit and license URL into description

### ISSUE THREAD

**Issue by artlogic** [arsmentis/wdet-conversion#69]
Change the import so that the credit field is used as the link text and the source/license is used as the href.

**Comment by artlogic** [comment]
@metastange I want to tag you here because this will result in a training issue - instead of having discrete fields, your authors will need to be adding proper HTML to asset descriptions, like so:

```html
<a href="https://wikipedia.org">Wikipedia</a>
```

**Comment by artlogic** [comment]
I believe there's a plugin that will allow us to handle this in a way that's closer to the old CMS: https://www.advancedcustomfields.com/

I'll be investigating this today.

**Comment by artlogic** [comment]
It looks like that plugin does allow us to have discrete fields. I'm working on getting this setup today.

**Comment by metastange** [comment]
Will the plugin eliminate the need for us to put HTML in the asset description? No worries either way, just trying to keep my training notes straight.

**Comment by artlogic** [comment]
> Will the plugin eliminate the need for us to put HTML in the asset description? No worries either way, just trying to keep my training notes straight.

Yes - the plugin will create an interface that's similar to the current CMS. It should be a fairly easy transition.

**Comment by artlogic** [comment]
Be sure to ignore source / license urls that aren't actually urls.

**Comment by artlogic** [comment]
Data is being included with assets via 821c380488e73ad1e3e7dda3762cf552dd5502c2

**Comment by artlogic** [comment]
Config is being brought in via 8afb1e8b69f53644987fa90debca24efa6d41739

This is now just a theming issue (getting credit/source to display in the front end, and *maybe* in the editor). Removing import issue tag.

**Comment by artlogic** [comment]
I expect this issue to be resolved by January 3rd.

**Comment by dorothyhernandez** [comment]
@wdet30 Credit/source still not displaying on the front end even though we are filling in the Credit field. I can figure out a workaround for the inline images but the featured images really need to have the credits showing up.

**Comment by artlogic** [comment]
I'm sorry this issue isn't resolved yet - please don't use workarounds as it may result in the credit/source being displayed twice. I will be working on this issue over the next few days.

**Comment by dorothyhernandez** [comment]
@artlogic copy that, thanks!

**Comment by artlogic** [comment]
I'm currently waiting to hear back from Advanced Custom Field plug-in support. Roughly speaking, the custom fields we set aren't available using the API I'd prefer to use - but they should be. Hopefully support can provide a work-around.

**Comment by artlogic** [comment]
The following will need to be added to functions.php to bring the proper fields into the REST API:


register_post_meta( 'attachment', 'credit', array(
    'type'         => 'string',
    'description'  => 'A meta key associated with a string meta value.',
    'single'       => true,
    'show_in_rest' => true,
) );

register_post_meta( 'attachment', 'source_license_url', array(
    'type'         => 'string',
    'description'  => 'A meta key associated with a string meta value.',
    'single'       => true,
    'show_in_rest' => true,
) );

Then data is available via https://wdetdev.wdet.org/wp-json/wp/v2/media/108960

```json
{
  meta: {
    credit: "credit name",
    source_license_url: "optional url"
  }
}
```
